### PR TITLE
Raw output prefix takes precedence for remote proxy creation in entrypoint

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
When running in GCP without a config file present, flytekit uses the wrong setting to try to determine what environment it's running in.  We should base the decision on the offloaded data location, not the cloud provider setting, which defaults to AWS.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Add an extra conditional to first check the `raw_output_data_prefix`.  Not too worried about the increased complexity becase this section of the code will soon be removed by #559, which should be going in within a week and change.

### CODE REVIEW

File: flytekit/bin/entrypoint.py [hunk 1]
```diff
@@ -226,7 +226,25 @@ def setup_execution(
         tmp_dir=user_workspace_dir,
     )
 
-    if cloud_provider == _constants.CloudProvider.AWS:
+    # This rather ugly condition will be going away with #559. We first check the raw output prefix, and if missing,
+    # we fall back to the logic of checking the cloud provider. The reason we have to check for the existence of the
+    # raw_output_data_prefix arg first is because it may be set to None by execute_task_cmd. That is there to support
+    # the corner case of a really old propeller that is still not filling in the raw output prefix template.
+    if raw_output_data_prefix:
```
Comment (kumare3) [inline flytekit/bin/entrypoint.py:233]: Can we put a comment that points
To the other PR and says this will be deleted

**Review by cosmicBboy — APPROVED** [review]

File: flytekit/bin/entrypoint.py (hunk 1 above)
Comment (wild-endeavor) [reply flytekit/bin/entrypoint.py:233]: that is already in the comment right above this line.
